pal/system-info/minigame/system-info.ts

```diff
@@ -97,12 +97,18 @@ class SystemInfo extends EventTarget {
         this.isXR = false;
 
         // init capability
+        // NOTE: canvas.toDataURL() is not supported on WeChat iOS end (Found on iPhone 7p)
+        const isIOSWechat = WECHAT && this.os === OS.IOS && !minigame.isDevTool;
         const _tmpCanvas1 = document.createElement('canvas');  // TODO: remove this
         let supportWebp;
-        try {
-            supportWebp = TEST ? false : _tmpCanvas1.toDataURL('image/webp').startsWith('data:image/webp');
-        } catch (e) {
-            supportWebp  = false;
+        if (isIOSWechat) {
+            supportWebp = true;
+        } else {
+            try {
+                supportWebp = TEST ? false : _tmpCanvas1.toDataURL('image/webp').startsWith('data:image/webp');
+            } catch (e) {
+                supportWebp  = false;
+            }
         }
```

Comment:
I think it is better to add a function named `supportsWebp`.

Reply:
OK